astroquery/mocserver/core.py

```diff
@@ -1,416 +1,707 @@
-    `regions.PolygonSkyRegion` or a `mocpy.MOC` object.
-
-    This package implements two methods:
-
-    * :meth:`~astroquery.mocserver.MOCServerClass.query_region` retrieving data-sets
-      (their associated MOCs and meta-datas) having sources in a given region.
-    * :meth:`~astroquery.mocserver.MOCServerClass.find_datasets` retrieving data-sets
-      (their associated MOCs and meta-datas) based on the values of their meta-datas.
+    """Query the `CDS MOCServer <http://alasky.unistra.fr/MocServer/query>`_.
 
+    The `CDS MOCServer <http://alasky.unistra.fr/MocServer/query>`_ allows the user
+    to retrieve all the datasets (with their meta-data) having sources in a specific
+    space-time region. This region can be a `regions.CircleSkyRegion`, a
+    `regions.PolygonSkyRegion`, a `mocpy.MOC`, a `mocpy.TimeMOC`, or a `mocpy.STMOC`
+    object.
     """
+
     URL = conf.server
     TIMEOUT = conf.timeout
+    DEFAULT_FIELDS = conf.default_fields
 
     def __init__(self):
         super().__init__()
-        self.path_moc_file = None
         self.return_moc = False
 
-    def query_region(self, *, region=None, get_query_payload=False, verbose=False, **kwargs):
-        """
-        Query the `CDS MOCServer <http://alasky.unistra.fr/MocServer/query>`_ with a region.
-
-        Can be a `regions.CircleSkyRegion`, `regions.PolygonSkyRegion` or `mocpy.MOC` object. Returns the data-sets
-        having at least one source in the region.
+    def query_region(
+        self, region=None,
+        *,
+        meta_data=None,
+        intersect="overlaps",
+        return_moc=None,
+        max_norder=None,
+        fields=None,
+        max_rec=None,
+        casesensitive=False,
+        spacesys=None,
+        get_query_payload=False,
+        verbose=False,
+        cache=True,
+    ):
+        """Query the MOC Server.
 
         Parameters
         ----------
-        region : `regions.CircleSkyRegion`, `regions.PolygonSkyRegion` or `mocpy.MOC`
-            The region to query the MOCServer with.
-            Can be one of the following types:
-
-            * ``regions.CircleSkyRegion`` : defines an astropy cone region.
-            * ``regions.PolygonSkyRegion`` : defines an astropy polygon region.
-            * ``mocpy.moc.MOC`` : defines a MOC from the MOCPy library. See the `MOCPy's documentation
-              <https://cds-astro.github.io/mocpy/>`__ for how to instantiate a MOC object.
-
+        region : `regions.CircleSkyRegion`, `regions.PolygonSkyRegion`, `mocpy.MOC`,
+            `mocpy.TimeMOC`, or `mocpy.STMOC`
+            The region to query the MOCServer with. Note that this can also be a
+            space-time region with the Time-MOCs and Space-Time-MOCs.
+            Defaults to None, which means that the search will be on the whole sky.
+        meta_data : str
+            Algebraic expression to select the datasets.
```

Comment:
I don't think `meta_data` is the right name here, at least it feels weird given the functionality. I don't have a good alternative suggestion though, maybe `dataset_filter`?

Also, is this algebraic? the `"ID=*HST*"` example feels more like a regex.

Reply:
OK, never mind, I see that this kwarg has already been used in the previous version, so keeping it makes sense.

Reply:
I kept it because it was already there, but I also dislike it. Maybe it's worth deprecating it now during the big change. What about `criteria` ? That'd be similar to what exists in SIMBAD and VizieR's modules. I'm not opposed to `dataset_filter` neither.

About the `Algebraic`, sure, I don't know why I left it there